src/query/storages/iceberg/src/catalog.rs

```diff
@@ -196,6 +198,29 @@ impl IcebergCatalog {
                 let ctl = RestCatalog::new(cfg);
                 Arc::new(ctl)
             }
+            IcebergCatalogOption::Glue(glue) => {
+                let cfg = GlueCatalogConfig::builder()
+                    .warehouse(glue.warehouse.clone())
+                    .props(
+                        glue.props
+                            .clone()
+                            .into_iter()
+                            .map(|(k, v)| (k.trim_matches('"').to_string(), v))
+                            .collect(),
+                    )
+                    .build();
+
+                // Due to the AWS Glue catalog creation being asynchronous, forced to run it a bit different way, so we don't have to make the outer function asynchronous.
+                let ctl = tokio::task::block_in_place(|| {
```

Comment:
However, this change might require a significant refactor. I'm happy to take it on if you're not interested.

Reply:
`tokio::task::block_in_place()` won't block the runtime, not like `futures::executor::block_on()`:

https://docs.rs/tokio/latest/tokio/task/fn.block_in_place.html
> Runs the provided blocking function on the current thread without blocking the executor.

It would be OK using it here.

Reply:
Thank you @drmingdrmer for pointing this out. I will review this section again. 💌

Reply:
Done.
<!-- Reviewable comment -OBfAYz--8m89uN-7Kll:b-896fix -->
<!-- Sent from Reviewable.io -->

Reply:
`databend_common_base::runtime::block_on` internally calls `tokio::task::block_in_place()`, there should not be the need to wrap it in another `tokio::task::block_in_place()` AFAIK.

@Xuanwo Is there any test that asserts the behavior of these introduced change?
<!-- Reviewable comment -OBh5cb44vuoU61xDXx9:bgeo6jn -->
<!-- Sent from Reviewable.io -->

Reply:
No such test I guess. I believe we can merge this PR first.

Reply:
Ok. Let's merge it once the double `block_in_place` issue is fixed.